presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -199,13 +199,20 @@ public Expression rewriteExpression(Expression node, Void context, ExpressionTre
             }
         }, expression);
 
+        // redo the analysis since above expression rewriter might change the type of some expressions
```

Comment:
This comment is not accurate. The issue is that the rewrite above creates new expression instances, so they don't have an entry in the type map.